plugins/destination/gremlin/client/spec.go

```diff
@@ -91,3 +109,108 @@ func (s *Spec) Validate() error {
 
 	return nil
 }
+
+func (Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
+	authMode := sc.Properties.Value("auth_mode")
+	authMode.Enum = []any{authModeNone, authModeBasic, authModeAWS}
+
+	sc.OneOf = []*jsonschema.Schema{
+		{
+			// if auth_mode is aws, require aws_region to be set
+			Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+				one := uint64(1)
+				properties := jsonschema.NewProperties()
+
+				authMode := *sc.Properties.Value("auth_mode")
+				authMode.Enum = []any{authModeAWS}
+				authMode.Pattern = string(authModeAWS)
+				properties.Set("auth_mode", &authMode)
+
+				awsRegion := *sc.Properties.Value("aws_region")
+				awsRegion.MinLength = &one
+				properties.Set("aws_region", &awsRegion)
+
+				return properties
+			}(),
+			Required: []string{"auth_mode", "aws_region"},
+		},
+		{
+			// If auth_mode is not aws, don't care about aws_region
+			Not: &jsonschema.Schema{
+				Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+					properties := jsonschema.NewProperties()
+					authMode := *sc.Properties.Value("auth_mode")
+					authMode.Enum = []any{authModeAWS}
+					authMode.Pattern = string(authModeAWS)
+					properties.Set("auth_mode", &authMode)
+					return properties
+				}(),
+				Required: []string{"auth_mode"},
+			},
+		},
+	}
+
+	sc.AllOf = []*jsonschema.Schema{
+		{
+			// if username is set, require password to be set and auth_mode to be basic
```

Comment:
shouldn't it be the other way around?
if mode is basic, require username & password.

`Validate` func suggests making `auth_mode:none => username & password should have max len = 0`

Reply:
It's the same thing but I'll rewrite it with const